Assets/Scripts/DrawLine.cs

```diff
@@ -25,27 +25,45 @@ void Awake()
         lineRenderer.endWidth = 0.15f;
     }
 
-    
+
 
     void Update()
     {
+        //testing to make sure all the waypoints can be drawn. Works!
+        //if(drawing) {
+        //    lineRenderer.SetPosition(0, waypoints[0]);
+        //    for(int i = 0; i < waypoints.Count; i++) {
+        //        lineRenderer.positionCount++;
+        //        Vector3 p = waypoints[i];
+        //        lineRenderer.SetPosition(i, p);
+        //    }
+        //}
+        //return;
+
         if (drawing)
         {
-            if (counter <= dist)
-            {
-                counter += (lineDrawSpeed) * Time.deltaTime;
-                lineRenderer.SetPosition(currentWaypoint, Vector3.Lerp(waypoints[currentWaypoint], waypoints[currentWaypoint + 1], counter));
+            counter += (lineDrawSpeed) * Time.deltaTime;
+
+            if(counter <= dist)
+            {   //counter starts at 0, so this is block is evaluated right off the start
+                lineRenderer.SetPosition(currentWaypoint+1, Vector3.Lerp(waypoints[currentWaypoint], waypoints[currentWaypoint + 1], counter/dist)); //lerp works on [0,1], counter is [0,dist] so you have to normalize it
+                //currentWaypoint+1, we offset by 1 because you already have currentWaypoint (0) set on the first iteration
+
                 front.transform.position = waypoints[currentWaypoint + 1];
-            }
+
+            } 
             else
             {
-                if (waypoints.Count > lineRenderer.positionCount + 1)
+                if(waypoints.Count > lineRenderer.positionCount) 
```

Comment:
Removing the +1, as it was stopping the line from rendering the last waypoint segment

Reply:
"stopping", I mean it was exiting out one waypoint too early.